### PR TITLE
Python: Optional dependencies

### PR DESCRIPTION
**Pull request by awharrison-28**
### Motivation and Context
The release of Python SK from 0.2.4 -> 0.2.6 significantly increased the size of the pip package. The dependencies causing this are not core to the kernel, so this PR establishes a pattern for SK connector dependencies (ex: Hugging Face, memory storage)

### Description
- Added a poetry group called hugging_face - this is the same name as the connectors namespace in the code.
- Added dependency installs to the Hugging Face sample notebook.
- removed Hugging Face dependencies from requirements.txt.
- Added import errors for torch, transformers, and sentence-transformers in the hugging_face connector classes.
- Added command `--with hugging_face` to integration test setup.

### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [ ] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: python/semantic_kernel/connectors/ai/hugging_face/services/hf_text_completion.py [hunk 1]
```diff
@@ -3,8 +3,8 @@
 from logging import Logger
 from typing import Optional
 
-import torch
-from transformers import GenerationConfig, pipeline
+# import torch
```
Comment (alexchaomander) [inline python/semantic_kernel/connectors/ai/hugging_face/services/hf_text_completion.py:6]: Should we leave these as commented out? Or can we just remove?

File: python/semantic_kernel/connectors/ai/hugging_face/services/hf_text_embedding.py [hunk 2]
```diff
@@ -2,10 +2,10 @@
 
 from logging import Logger
 from typing import List, Optional
-
-import torch
 from numpy import array, ndarray
-from sentence_transformers import SentenceTransformer
+
+# import torch
```
Comment (alexchaomander) [inline python/semantic_kernel/connectors/ai/hugging_face/services/hf_text_embedding.py:7]: Same, necessary to leave comments here?

**Review by alexchaomander — DISMISSED** [review]

File: python/semantic_kernel/connectors/ai/hugging_face/services/hf_text_completion.py [hunk 3]
```diff
@@ -3,8 +3,8 @@
...
+# from transformers import GenerationConfig, pipeline
```
Comment (shawncal) [inline python/semantic_kernel/connectors/ai/hugging_face/services/hf_text_completion.py:7]:
```suggestion
```

File: python/semantic_kernel/connectors/ai/hugging_face/services/hf_text_embedding.py [hunk 4]
```diff
@@ -2,10 +2,10 @@
...
+# from sentence_transformers import SentenceTransformer
```
Comment (shawncal) [inline python/semantic_kernel/connectors/ai/hugging_face/services/hf_text_embedding.py:8]:
```suggestion
```

**Review by shawncal — APPROVED** [review]

**Pull request merged**